numpy/lib/_iotools.py

```diff
@@ -189,12 +189,10 @@ def autostrip(self, method):
         return lambda input: [_.strip() for _ in method(input)]
     #
 
-    def __init__(self, delimiter=None, comments=b'#', autostrip=True):
+    def __init__(self, delimiter=None, comments='#', autostrip=True, encoding=None):
```

Comment:
Should this be `u'#'`?

Reply:
Hmm, looks like it if we work in unicode. The `u` prefix came in late (python 3.4 ?), but is now available in all the versions we support.

Reply:
I think we are OK, in Python 2.7
```
In [1]: u'123' + '4'
Out[1]: u'1234'

In [2]: '3' in u'123'
Out[2]: True
```
So the strings are promoted to unicode. Narrow builds (ucf-2 unicode), might be a problem.